ntbs-service/Models/Entities/Notification.cs

```diff
@@ -127,19 +126,25 @@ public Notification()
                                        || NotificationStatus == NotificationStatus.Closed
                                        || NotificationStatus == NotificationStatus.Legacy;
 
-        [AssertThat(
-            @"ShouldValidateFull && (ClinicalDetails.IsPostMortem != true || IsPostMortemAndHasCorrectEvents)",
-            ErrorMessage = ValidationMessages.IsPostMortemButTreatmentEventsDoNotMatch)]
+        [AssertThat(@"ShouldValidateFull && HasDeathEventForPostMortemCase", ErrorMessage = ValidationMessages.DeathEventRequiredForPostMortemCase)]
+        public bool HasDeathEventForPostMortemCase =>
```

Comment:
I think this should be called `HasDeathEventIfPostMortemCase` since it's true for non-post-mortem cases